plugins/languages/cmake-tools.nix

```diff
@@ -0,0 +1,391 @@
+        '';
+
+        opts = helpers.defaultNullOpts.mkAttributeSet "{}" ''
+          The options the executor will get, possible values depend on the executor type. See `default_opts` for possible values.
+        '';
+
+        default_opts =
+          helpers.defaultNullOpts.mkAttributeSet ''
+            {
+              quickfix = {
+                show = "always"; # "always", "only_on_error"
+                position = "belowright"; # "vertical", "horizontal", "leftabove", "aboveleft", "rightbelow", "belowright", "topleft", "botright", use `:h vertical` for example to see help on them
+                size = 10;
+                encoding = "utf-8"; # if encoding is not "utf-8", it will be converted to "utf-8" using `vim.fn.iconv`
+                auto_close_when_success = true; # typically, you can use it with the "always" option; it will auto-close the quickfix buffer if the execution is successful.
+              };
+
+              toggleterm = {
+                direction = "float"; # 'vertical' | 'horizontal' | 'tab' | 'float'
+                close_on_exit = false; # whether close the terminal when exit
+                auto_scroll = true; # whether auto scroll to the bottom
+              };
+
+              overseer = {
+                new_task_opts = {
+                  strategy.__raw = '''
+                    {
+                      "terminal",
+                    },
+                  '''';
+                }; # options to pass into the `overseer.new_task` command
+                on_new_task = {
+                  __raw = '''
+                    function(task) end
+                  ''';
+                }; # a function that gets overseer.Task when it is created, before calling `task:start`
+              };
+
+              terminal = {
+                name = "Main Terminal";
+                prefix_name = "[CMakeTools]: "; # This must be included and must be unique, otherwise the terminals will not work. Do not use a simple spacebar " ", or any generic name
+                split_direction = "horizontal"; # "horizontal", "vertical"
+                split_size = 11;
+
+                # Window handling
+                single_terminal_per_instance = true; # Single viewport, multiple windows
+                single_terminal_per_tab = true; # Single viewport per tab
+                keep_terminal_static_location = true; # Static location of the viewport if available
+
+                # Running Tasks
+                start_insert = false; # If you want to enter terminal with :startinsert upon using :CMakeRun
+                focus = false; # Focus on terminal when cmake task is launched.
+              }; # terminal executor uses the values in cmake_terminal
+            }
+          ''
+          ''
+            A list of default and possible values for executors.
+          '';
+      };
+
+    cmake_runner =
+      helpers.mkCompositeOption ''
+        Runner to use.
+      ''
+      {
```

Comment:
Again, prefer not using composite options, but you can if you think the description is necessary:

```nix
cmake_runner = {
```